flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -280,6 +294,31 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
 	}
 
+	/**
+	 * Registers legacy network metric groups before shuffle service refactoring.
+	 *
+	 * <p>Registers legacy metric groups if shuffle service implementation is original default one.
+	 *
+	 * @deprecated should be removed in future
+	 */
+	@SuppressWarnings("DeprecatedIsStillUsed")
+	@Deprecated
+	public void registerLegacyNetworkMetrics(
+		MetricGroup metricGroup,
+		ResultPartitionWriter[] producedPartitions,
+		InputGate[] inputGates) {
+		// add metrics for buffers
+		final MetricGroup buffersGroup = metricGroup.addGroup("buffers");
+
+		// similar to MetricUtils.instantiateNetworkMetrics() but inside this IOMetricGroup
```

Comment:
It seems we could not see `IOMetricGroup` here, might need adjust the annotation.

Reply:
the comment is still relevant but I will put `metricGroup` in parenthesis:
`.. inside this IOMetricGroup (metricGroup)`